### PR TITLE
[1726] Order keys in nrepl message objects.

### PR DESCRIPTION
**Pull request by harold**
Part of the overall solution described in issue 1726.

In order to have the keys print in a predictable order, we can sort them before printing.

Example before this change:
```
(-->
  ns  "cortex.dataset"
  op  "eval"
  session  "499ca8d8-43c8-40bf-aa19-ff369673ce4c"
  code  "(defn map-sequence->dataset "Turns an infinite training se..."
  file  "/home/harold/src/cortex/src/cortex/dataset.clj"
  line  377
  column  1
  id  "37"
)
```

Example after this change:
```
(-->
  code  "(defn map-sequence->dataset "Turns an infinite training se..."
  column  1
  file  "/home/harold/src/cortex/src/cortex/dataset.clj"
  id  "45"
  line  377
  ns  "cortex.dataset"
  op  "eval"
  session  "499ca8d8-43c8-40bf-aa19-ff369673ce4c"
)
```

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our contribution guidelines
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] You've updated the changelog (if adding/changing user-visible functionality)

Thanks!

### CODE REVIEW

**Comment by harold** [comment]
Hello @bbatsov --- inspired by your tweet: https://twitter.com/bbatsov/status/839409503632261121

Some more work on: https://github.com/clojure-emacs/cider/issues/1726

I haven't done _that_ much elisp before, so there may be a better way, but it works for me in simple tests.

**Comment by bbatsov** [comment]
Thanks for tackling this!

`id`, `op` and `session` have special importance as they are part of every request and response. They should always be the 3 first keys that get printed and the rest can be sorted alphabetically.

File: nrepl-client.el [hunk 1]
```diff
@@ -1219,11 +1219,16 @@ FOREGROUND and BUTTON are as in `nrepl-log-pp-object'."
     (let ((head (format "(%s" (car object))))
       (insert (color head))
       (let ((indent (+ 2 (- (current-column) (length head))))
-            (l (point)))
+            (l (point))
+            (sorted-object (if (cdr object)
+                               (apply 'seq-concatenate 'list
+                                      (sort (seq-partition (copy-list (cdr object)) 2)
```
Comment (bbatsov) [inline nrepl-client.el]: You should use `cl-copy-list`.

**Comment by harold** [comment]
That all makes sense --- I'll do another pass on this tomorrow. I'm out of time for today.

**Comment by harold** [comment]
@bbatsov --- got `id`, `op`, and `session` to print first. I also snuck in the table alignment:

```
(-->
  id      "67"
  op      "eval"
  session "d9131a55-30bf-405c-ac9e-ed4c90be9ad4"
  code    "(defmulti loss
  "Implement a specific loss based on the typ..."
  column  1
  file    "/home/harold/src/cortex/src/cortex/loss.clj"
  line    136
  ns      "cortex.loss"
)```

**Comment by bbatsov** [comment]
Looks great! I was reminded of just one more thing - now we have `time-stamp` which is also special. We should put it either first or after `session` IMO.

**Comment by harold** [comment]
Is that new? I haven't seen it (I'm still running the latest MELPA release).

Adding a string here:
https://github.com/clojure-emacs/cider/pull/1956/files#diff-bc7f0008c444f0fbf63329a565cdf64fR1230

Should be pretty easy, if you want to make any other names special. Notably, the special keys also appear in alphabetical order, so `time-stamp` after `session` would be easiest. ;)

**Comment by bbatsov** [comment]
Pretty new - https://github.com/clojure-emacs/cider/commit/050b2a7065c0a278eaf500f9f7b2a081893989fb

Yeah, I saw adding this would be easy. Just thought now would be the best time to do it.

**Comment by harold** [comment]
@bbatsov --- I'll be out till Monday (tomorrow is my birthday).

Perhaps you'd like to pull this into your repo and add that and merge? Otherwise I can check it out Monday.

Regardless, thanks, this was fun and enlightening. Hacking on Cider is easier than I thought it might be.

**Comment by bbatsov** [comment]
> Perhaps you'd like to pull this into your repo and add that and merge? Otherwise I can check it out Monday.

No worries! I'll wrap this my myself.

> Regardless, thanks, this was fun and enlightening. Hacking on Cider is easier than I thought it might be.

Happy to hear this! Hopefully I'll be seeing more of you here! ;-)

Happy birthday!

**Pull request merged**